spec/helpers/ajax-helper.ts

```diff
@@ -218,4 +226,33 @@ export class MockXMLHttpRequest {
       }
     });
   }
-}
\ No newline at end of file
+}
+
+export class MockXMLHttpRequestInternetExplorer extends MockXMLHttpRequest {
+  constructor() {
+    super();
+  }
+
+  private mockHttp204() {
+    this.responseType = '';
+    this.responseText = '';
+    this.response = '';
+  }
+
+  protected jsonResponseValue(response: any) {
+    if (this.status == 204) {
+      this.mockHttp204();
+      return;
+    }
+    return super.jsonResponseValue(response);
+  }
+
+  protected defaultResponseValue() {
```

Comment:
> That is why when we mock IE that we actually unset responseType in private mockHttp204()

: yes, that I'm aware of.

My question in here is, if overrided `defaultResponseValue` doesn't do any roles in inherited `MockXMLHttpRequestIE`, is it necessarily need to do it? test for behavior of IE is fulfilled by response override at line 242, so default case may not be separated / overrideded but just leave as-is to throw directly in line 199 instead. Of course it's for 'just in case' and to be possibly served in further though, I'm seeing inherited class can be simplified.